src/scraper/scraper.cpp

```diff
@@ -2230,57 +2230,69 @@ bool ProcessProjectRacFileByCPID(const std::string& project, const fs::path& fil
+                break;
+            }
 
-                    const std::string username = ExtractXML(data, "<name>", "</name>");
+            bool just_verified = false;
+            if (!already_verified)
+            {
+                // Attempt to verify.
 
-                    // Base58-encoded beacon verification code sizes fall within:
-                    if (username.size() < 26 || username.size() > 28) continue;
+                const std::string username = ExtractXML(data, "<name>", "</name>");
 
+                // Base58-encoded beacon verification code sizes fall within:
+                if (username.size() >= 26 && username.size() <= 28)
+                {
                     // The username has to be temporarily changed to a "verification code" that is
                     // a base58 encoded version of the public key of the pending beacon, so that
                     // it will match the mPendingMap entry. This is the crux of the user validation.
                     const auto iter_pair = Consensus.mPendingMap.find(username);
 
-                    if (iter_pair == Consensus.mPendingMap.end()) continue;
-                    if (iter_pair->second.cpid != cpid) continue;
-
-                    // This copies the pending beacon entry into the local VerifiedBeacons map and updates
-                    // the time entry.
-                    IncomingVerifiedBeacons.mVerifiedMap[iter_pair->first] = iter_pair->second;
+                    if (iter_pair != Consensus.mPendingMap.end() && iter_pair->second.cpid == cpid)
+                    {
+                        // This copies the pending beacon entry into the local VerifiedBeacons map and updates
+                        // the time entry.
+                        IncomingVerifiedBeacons.mVerifiedMap[iter_pair->first] = iter_pair->second;
 
-                    _log(logattribute::INFO, "ProcessProjectRacFileByCPID", "Verified pending beacon for verification code "
-                         + iter_pair->first + ", cpid " + iter_pair->second.cpid);
+                        _log(logattribute::INFO, "ProcessProjectRacFileByCPID", "Verified pending beacon for verification code "
+                             + iter_pair->first + ", cpid " + iter_pair->second.cpid);
 
-                    IncomingVerifiedBeacons.timestamp = GetAdjustedTime();
+                        IncomingVerifiedBeacons.timestamp = GetAdjustedTime();
 
-                    // We do NOT want to add a just verified CPID to the statistics this iteration,
-                    // because we may be halfway through processing the set of projects. Instead, add to the
-                    // incoming verification map (above), which will be handled in the calling function once
-                    // all of the projects are gone through. This will become a verified beacon the next time
-                    // around.
-                    continue;
+                        just_verified = true;
```

Comment:
Do we need to check `just_verified` somewhere? It looks like it's unused.

Reply:
Actually we don't, according to my truth table.

Reply:
Removed the flag and simplified the truth table.